[MLIR][NVVM] Combine griddepcontrol Ops

grypp opened the pull request:
We've 2 ops:
1. nvvm.griddepcontrol.wait
2. nvvm.griddepcontrol.launch_dependents

They are related to Grid Dependent Launch (or programmatic dependent launch in CUDA) and same concept. This PR unifies both ops into a single one.

schwarzschild-radius commented on mlir/test/Target/LLVMIR/nvvmir.mlir:
```diff
@@ -918,3 +918,14 @@ llvm.func @nvvm_dot_accumulate_2way(%a: vector<2xi16>, %b: vector<4xi8>, %c: i32
+// -----
+
+// CHECK-LABEL: @nvvm_pmevent
+llvm.func @nvvm_pmevent() {
```
I think this diff is unrelated to the current PR?

schwarzschild-radius reviewed (approved):
Thanks for the change! Just a couple of minor nits, LGTM!

Wolfram70 commented on mlir/include/mlir/Dialect/LLVMIR/NVVMOps.td:
```diff
@@ -2995,30 +2995,47 @@ def NVVM_WgmmaMmaAsyncOp : NVVM_Op<"wgmma.mma_async",
+def GridDepActionWait : I32EnumAttrCase<"wait", 0>;
+def GridDepActionLaunchDependent : I32EnumAttrCase<"launch_dependents", 1>;
 
+def GridDepActionKind : I32EnumAttr<"GridDepActionKind", "Action kind for grid dependency control",
```
nit: Maybe we can use `I32EnumCase` and `I32Enum` here (added in https://github.com/llvm/llvm-project/pull/132148)?

grypp committed:
We've 2 ops:
1. nvvm.griddepcontrol.wait
1. nvvm.griddepcontrol.launch_dependents

They are related to FDL and same concept. This PR unifies both ops into a single one.
